2000 Maryland Congressional Districts

KT0416838 opened the pull request:
## Redistricting requirements
In ``Maryland``, according to [NCSL Redistricting Law 2000](https://web.archive.org/web/20041216185957/https://www.senate.mn/departments/scr/redist/red2000/Tab5appx.htm), districts should:

1. be contiguous
1. have equal populations
1. be geographically compact
1. preserve county and municipality boundaries as much as possible
1. not dilute minority voting strengths
1. give due regard to natural boundaries

### Algorithmic Constraints
We enforce a maximum population deviation of 0.5%.

## Data Sources
Data for ``Maryland`` comes from the [ALARM Project's update](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/ZV5KF3) to [The Record of American Democracy](https://road.hmdc.harvard.edu/).

## Pre-processing Notes
No manual pre-processing decisions were necessary.

## Simulation Notes
We sample 10,000 districting plans for ``Maryland`` across 5 independent runs of the SMC algorithm.
We then thinned the number of samples to 5,000. 
No special techniques were needed to produce the sample.

## Validation
<img width="3000" height="4500" alt="validation_20250801_1501" src="https://github.com/user-attachments/assets/b3360b6b-d080-432f-beb0-ad283df9c6ff" />

```
✔ Creating <redist_map> object for MD_cd_2000 ... done
SMC: 5,000 sampled plans of 8 districts on 1,691 units
`adapt_k_thresh`=0.99 • `seq_alpha`=0.5
`pop_temper`=0                     
ℹ Downloading files for WV_cd_2000

Plan diversity 80% range: 0.48 to 0.73
ℹ Downloading files for WV_cd_2000

R-hat values for summary statistics:
  pop_overlap     total_vap      plan_dev     comp_edge   comp_polsby     pop_other       pop_two 
        1.004         1.000         1.002         1.001         1.011         1.011         1.007 
     pop_aian     pop_white     pop_black      pop_hisp     pop_asian      pop_nhpi      vap_hisp 
        1.006         1.005         1.005         1.005         1.005         1.003         1.005 
     vap_aian     vap_asian     vap_white     vap_black      vap_nhpi     vap_other       vap_two 
        1.005         1.005         1.005         1.005         1.004         1.009         1.007 
county_splits   muni_splits           ndv           nrv       ndshare 
        1.012         1.002         1.007         1.009         1.009 

Sampling diagnostics for SMC run 1 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,950 (97.5%)     15.2%        0.30 1,268 (100%)      5 
Split 2     1,929 (96.4%)     13.6%        0.36 1,226 ( 97%)      5 
Split 3     1,905 (95.2%)     12.0%        0.41 1,224 ( 97%)      5 
Split 4     1,853 (92.6%)     17.5%        0.50 1,204 ( 95%)      3 
Split 5     1,789 (89.5%)     14.7%        0.60 1,183 ( 94%)      3 
Split 6     1,772 (88.6%)     11.4%        0.65 1,128 ( 89%)      3 
Split 7     1,776 (88.8%)      5.4%        0.66 1,033 ( 82%)      2 
Resample    1,193 (59.6%)       NA%        0.66 1,463 (116%)     NA 

Sampling diagnostics for SMC run 2 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,951 (97.5%)     12.8%        0.30 1,266 (100%)      6 
Split 2     1,934 (96.7%)     13.7%        0.35 1,271 (101%)      5 
Split 3     1,903 (95.2%)     14.9%        0.42 1,220 ( 97%)      4 
Split 4     1,868 (93.4%)     17.0%        0.49 1,217 ( 96%)      3 
Split 5     1,833 (91.7%)     11.3%        0.53 1,195 ( 95%)      4 
Split 6     1,782 (89.1%)     12.1%        0.63 1,156 ( 91%)      3 
Split 7     1,763 (88.1%)      6.0%        0.68 1,042 ( 82%)      2 
Resample    1,182 (59.1%)       NA%        0.68 1,443 (114%)     NA 

Sampling diagnostics for SMC run 3 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,950 (97.5%)     13.4%        0.30 1,263 (100%)      6 
Split 2     1,932 (96.6%)     17.1%        0.35 1,202 ( 95%)      4 
Split 3     1,902 (95.1%)     12.2%        0.42 1,232 ( 97%)      5 
Split 4     1,875 (93.8%)     17.8%        0.47 1,214 ( 96%)      3 
Split 5     1,845 (92.2%)     14.3%        0.53 1,194 ( 94%)      3 
Split 6     1,791 (89.6%)     17.2%        0.63 1,163 ( 92%)      2 
Split 7     1,790 (89.5%)      3.7%        0.65 1,041 ( 82%)      3 
Resample    1,281 (64.1%)       NA%        0.65 1,484 (117%)     NA 

Sampling diagnostics for SMC run 4 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,948 (97.4%)     12.9%        0.30 1,256 ( 99%)      6 
Split 2     1,929 (96.4%)     12.9%        0.36 1,234 ( 98%)      5 
Split 3     1,908 (95.4%)     14.7%        0.41 1,219 ( 96%)      4 
Split 4     1,884 (94.2%)     17.5%        0.46 1,220 ( 97%)      3 
Split 5     1,818 (90.9%)     11.2%        0.53 1,220 ( 97%)      4 
Split 6     1,782 (89.1%)     11.3%        0.63 1,178 ( 93%)      3 
Split 7     1,751 (87.5%)      4.0%        0.69 1,018 ( 81%)      3 
Resample    1,133 (56.7%)       NA%        0.69 1,409 (111%)     NA 

Sampling diagnostics for SMC run 5 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,950 (97.5%)     11.1%        0.30 1,290 (102%)      7 
Split 2     1,932 (96.6%)     13.4%        0.35 1,251 ( 99%)      5 
Split 3     1,912 (95.6%)     12.0%        0.40 1,233 ( 98%)      5 
Split 4     1,885 (94.3%)     13.7%        0.46 1,220 ( 97%)      4 
Split 5     1,835 (91.8%)     14.6%        0.54 1,197 ( 95%)      3 
Split 6     1,824 (91.2%)     16.9%        0.58 1,177 ( 93%)      2 
Split 7     1,803 (90.1%)      5.4%        0.63 1,072 ( 85%)      2 
Resample    1,321 (66.1%)       NA%        0.63 1,473 (117%)     NA 

•  Watch out for low effective samples, very low acceptance rates (less than 1%), large std. devs.
of the log weights (more than 3 or so), and low numbers of unique plans. R-hat values for summary
statistics should be between 1 and 1.05.
```

## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the main branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@christopherkenny

christopherkenny commented on analyses/MD_cd_2000/03_sim_MD_cd_2000.R:
```diff
@@ -0,0 +1,34 @@
+
+set.seed(2000)
+plans <- redist_smc(map, nsims = 2e3, runs = 5, counties = county)
+# IF CORES OR OTHER UNITS HAVE BEEN MERGED:
```
Can you remove these too?

KT0416838 replied:
Yes, I removed them.

KT0416838 commented:
1. I edited the adjacency graph so districts can’t traverse the Bay. Specifically, I excluded the Bay polygons from the graph. One island became isolated as a result, so I manually connected it to its closest neighbor. Validation looks good: the maps show East/West shores separated, and the summary shows no warning messages. I will walk through what I did at our next meeting so that we can decide whether to adopt this update.

<img width="3000" height="4500" alt="validation_20250810_1950" src="https://github.com/user-attachments/assets/2d893004-6b87-4726-8532-8a13ffe167df" />

```
✔ Creating <redist_map> object for MD_cd_2000 ... done
SMC: 5,000 sampled plans of 8 districts on 1,685 units
`adapt_k_thresh`=0.99 • `seq_alpha`=0.5       
`pop_temper`=0                                
ℹ Creating <redist_map> object for MD_cd_2000

Plan diversity 80% range: 0.37 to 0.61        
ℹ Creating <redist_map> object for MD_cd_2000

R-hat values for summary statistics:
  pop_overlap     total_vap      plan_dev     comp_edge   comp_polsby 
        1.005         1.007         1.003         1.004         1.004 
    pop_other       pop_two      pop_aian     pop_white     pop_black 
        1.003         1.007         1.004         1.001         1.001 
     pop_hisp     pop_asian      pop_nhpi      vap_hisp      vap_aian 
        1.002         1.004         1.002         1.003         1.002 
    vap_asian     vap_white     vap_black      vap_nhpi     vap_other 
        1.004         1.004         1.001         1.003         1.005 
      vap_two county_splits   muni_splits           ndv           nrv 
        1.005         1.004         1.003         1.005         1.006 
      ndshare 
        1.010 

Sampling diagnostics for SMC run 1 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,910 (95.5%)     14.0%        0.42 1,230 ( 97%)      6 
Split 2     1,903 (95.2%)     11.8%        0.43 1,215 ( 96%)      6 
Split 3     1,883 (94.1%)     12.6%        0.46 1,210 ( 96%)      5 
Split 4     1,839 (92.0%)     13.6%        0.54 1,227 ( 97%)      4 
Split 5     1,786 (89.3%)     16.0%        0.63 1,201 ( 95%)      3 
Split 6     1,756 (87.8%)     17.2%        0.68 1,175 ( 93%)      2 
Split 7     1,742 (87.1%)      3.8%        0.70 1,017 ( 80%)      3 
Resample    1,113 (55.6%)       NA%        0.70 1,423 (113%)     NA 

Sampling diagnostics for SMC run 2 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,911 (95.6%)     12.2%        0.41 1,269 (100%)      7 
Split 2     1,906 (95.3%)     14.1%        0.42 1,238 ( 98%)      5 
Split 3     1,873 (93.6%)     15.7%        0.47 1,234 ( 98%)      4 
Split 4     1,834 (91.7%)     17.5%        0.54 1,216 ( 96%)      3 
Split 5     1,793 (89.6%)     22.3%        0.62 1,228 ( 97%)      2 
Split 6     1,772 (88.6%)     16.9%        0.66 1,159 ( 92%)      2 
Split 7     1,697 (84.9%)      5.4%        0.74 1,065 ( 84%)      2 
Resample      982 (49.1%)       NA%        0.74 1,371 (108%)     NA 

Sampling diagnostics for SMC run 3 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,912 (95.6%)     14.1%        0.41 1,281 (101%)      6 
Split 2     1,914 (95.7%)     14.6%        0.41 1,232 ( 97%)      5 
Split 3     1,869 (93.5%)     21.0%        0.47 1,213 ( 96%)      3 
Split 4     1,812 (90.6%)     24.7%        0.56 1,243 ( 98%)      2 
Split 5     1,788 (89.4%)     15.8%        0.62 1,189 ( 94%)      3 
Split 6     1,772 (88.6%)     11.8%        0.66 1,160 ( 92%)      3 
Split 7     1,702 (85.1%)      4.1%        0.76 1,041 ( 82%)      3 
Resample    1,045 (52.2%)       NA%        0.76 1,356 (107%)     NA 

Sampling diagnostics for SMC run 4 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,915 (95.7%)     11.5%        0.41 1,220 ( 97%)      7 
Split 2     1,911 (95.6%)     13.8%        0.41 1,196 ( 95%)      5 
Split 3     1,871 (93.6%)     16.1%        0.47 1,226 ( 97%)      4 
Split 4     1,824 (91.2%)     18.4%        0.56 1,224 ( 97%)      3 
Split 5     1,782 (89.1%)     10.0%        0.62 1,211 ( 96%)      5 
Split 6     1,742 (87.1%)     12.4%        0.70 1,159 ( 92%)      3 
Split 7     1,699 (84.9%)      5.9%        0.74 1,035 ( 82%)      2 
Resample      998 (49.9%)       NA%        0.74 1,356 (107%)     NA 

Sampling diagnostics for SMC run 5 of 5 (2,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,910 (95.5%)     17.5%        0.42 1,263 (100%)      5 
Split 2     1,916 (95.8%)     17.7%        0.40 1,226 ( 97%)      4 
Split 3     1,894 (94.7%)     16.0%        0.45 1,221 ( 97%)      4 
Split 4     1,850 (92.5%)     18.5%        0.52 1,214 ( 96%)      3 
Split 5     1,789 (89.4%)     16.2%        0.63 1,196 ( 95%)      3 
Split 6     1,799 (89.9%)      8.9%        0.63 1,143 ( 90%)      4 
Split 7     1,732 (86.6%)      4.0%        0.69 1,046 ( 83%)      3 
Resample    1,050 (52.5%)       NA%        0.69 1,409 (111%)     NA 

•  Watch out for low effective samples, very low acceptance rates (less than 1%),
large std. devs. of the log weights (more than 3 or so), and low numbers of
unique plans. R-hat values for summary statistics should be between 1 and 1.05.
```

2. I found 86 discontiguous precincts, as shown below. I think this explains the discontiguous district in the middle map. The attached picture is an example of one precinct with four parts. Should I mention these discontiguous precincts in the documentation?

```
# A tibble: 8 × 2
  cd_2000 n_disc_vtds
    <int>       <int>
1       6          38
2       4          14
3       3          10
4       7          10
5       1           8
6       8           4
7       2           1
8       5           1
```

<img width="664" height="519" alt="Zoom 2401305-002 pieces vs  district boundaries" src="https://github.com/user-attachments/assets/e7c4807a-f288-444c-a77d-101845a57f35" />

KT0416838 commented:
I updated the PR based on our discussion. While all polygons are preserved, the adjacency graph looks appropriate to prevent districts from crossing the bay. Additionally, the validation maps look fine. For example, the left and right maps each have seven districts, whereas the enacted plan had eight. The same color is used for two different districts located on the east and west sides of the bay. Although the middle map appears to have eight districts, that may be due to discontiguous precincts.

<img width="1163" height="661" alt="Screenshot 2025-10-05 at 23 41 15" src="https://github.com/user-attachments/assets/bfcc3dc7-c25e-414e-af3e-a3ad3c5acc05" />

KT0416838 commented:
@christopherkenny, I’ve updated the code to include the `check_valid` function and the PR description. It now produces only contiguous plans. The districts don't traverse the bay, and R-hat values are also below 1.05. Could you please review them?

<img width="3000" height="4500" alt="validation_20251018_1649" src="https://github.com/user-attachments/assets/7c4678d3-1697-4962-9779-3bd7f10f0d82" />

```
SMC: 5,000 sampled plans of 8 districts on 1,691 units
Plans sampled on Graph Space using the Naive Top K forward kernel.
Forward kernel parameters: `adapt_k_thresh`=0.99
SMC Parameters:
• `weight_type` = optimal
• `seq_alpha` = 1
• `pop_temper` = 0
Plan diversity 80% range: 0.29 to 0.54
Largest R-hat values for ordered summary statistics:
    comp_edge   comp_polsby county_splits   muni_splits       ndshare 
        1.007         1.005         1.004         1.004         1.005 
          ndv           nrv      plan_dev      pop_aian     pop_asian 
        1.004         1.004         1.001         1.005         1.005 
    pop_black      pop_hisp      pop_nhpi     pop_other   pop_overlap 
        1.006         1.007         1.006         1.006         1.003 
      pop_two     pop_white     total_vap      vap_aian     vap_asian 
        1.005         1.006         1.004         1.005         1.006 
    vap_black      vap_hisp      vap_nhpi     vap_other       vap_two 
        1.006         1.007         1.007         1.006         1.005 
    vap_white 
        1.007 
• R-hat ≤ 1.05: 208
• 1.05 < R-hat ≤ 1.1: 0
• R-hat > 1.1: 0
• Total R-hats: 208
Sampling diagnostics for SMC run 1 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,830 (55.2%)     14.1%        0.85 28,471 (100%)      6 
Split 2    28,866 (64.1%)     15.6%        0.50 24,513 ( 86%)      4 
Split 3    17,962 (39.9%)     19.0%        0.69 25,831 ( 91%)      3 
Split 4    30,510 (67.8%)     16.7%        0.54 22,844 ( 80%)      3 
Split 5    24,211 (53.8%)     11.1%        0.52 25,652 ( 90%)      4 
Split 6    22,905 (50.9%)     11.0%        0.59 24,545 ( 86%)      3 
Split 7    43,884 (97.5%)      5.4%        0.16 22,052 ( 78%)      2 
Resample   43,884 (97.5%)       NA%          NA 42,132 (148%)     NA 

Sampling diagnostics for SMC run 2 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,855 (55.2%)     13.9%        0.84 28,415 (100%)      6 
Split 2    29,010 (64.5%)     12.6%        0.50 24,459 ( 86%)      5 
Split 3    17,852 (39.7%)     14.4%        0.69 25,815 ( 91%)      4 
Split 4    30,360 (67.5%)     16.8%        0.54 22,834 ( 80%)      3 
Split 5    24,553 (54.6%)     11.2%        0.52 25,621 ( 90%)      4 
Split 6    22,582 (50.2%)     11.0%        0.59 24,559 ( 86%)      3 
Split 7    43,899 (97.6%)      5.4%        0.16 21,945 ( 77%)      2 
Resample   43,899 (97.6%)       NA%          NA 42,150 (148%)     NA 

Sampling diagnostics for SMC run 3 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,730 (55.0%)     11.9%        0.85 28,275 ( 99%)      7 
Split 2    28,772 (63.9%)     12.5%        0.50 24,434 ( 86%)      5 
Split 3    17,916 (39.8%)     14.4%        0.69 25,824 ( 91%)      4 
Split 4    30,463 (67.7%)     16.7%        0.54 22,890 ( 80%)      3 
Split 5    24,089 (53.5%)     21.2%        0.52 25,609 ( 90%)      2 
Split 6    22,736 (50.5%)     16.0%        0.60 24,429 ( 86%)      2 
Split 7    43,905 (97.6%)      5.4%        0.16 21,993 ( 77%)      2 
Resample   43,905 (97.6%)       NA%          NA 42,123 (148%)     NA 

Sampling diagnostics for SMC run 4 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,883 (55.3%)     13.9%        0.85 28,169 ( 99%)      6 
Split 2    29,341 (65.2%)     12.5%        0.50 24,527 ( 86%)      5 
Split 3    17,857 (39.7%)     14.4%        0.68 25,950 ( 91%)      4 
Split 4    30,474 (67.7%)     16.7%        0.54 22,876 ( 80%)      3 
Split 5    24,063 (53.5%)     11.3%        0.52 25,731 ( 90%)      4 
Split 6    22,130 (49.2%)     11.0%        0.61 24,366 ( 86%)      3 
Split 7    43,930 (97.6%)      5.4%        0.16 21,988 ( 77%)      2 
Resample   43,930 (97.6%)       NA%          NA 42,207 (148%)     NA 

Sampling diagnostics for SMC run 5 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,877 (55.3%)     14.0%        0.85 28,262 ( 99%)      6 
Split 2    29,203 (64.9%)     12.5%        0.50 24,527 ( 86%)      5 
Split 3    17,839 (39.6%)     11.6%        0.69 25,869 ( 91%)      5 
Split 4    30,780 (68.4%)     12.8%        0.54 22,774 ( 80%)      4 
Split 5    24,316 (54.0%)     15.0%        0.52 25,737 ( 90%)      3 
Split 6    22,751 (50.6%)     15.9%        0.59 24,595 ( 86%)      2 
Split 7    43,910 (97.6%)      5.3%        0.16 22,096 ( 78%)      2 
Resample   43,910 (97.6%)       NA%          NA 42,201 (148%)     NA 

Sampling diagnostics for SMC run 6 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,751 (55.0%)     12.0%        0.85 28,224 ( 99%)      7 
Split 2    29,211 (64.9%)     12.6%        0.50 24,459 ( 86%)      5 
Split 3    17,747 (39.4%)     14.4%        0.69 25,800 ( 91%)      4 
Split 4    30,477 (67.7%)     16.9%        0.55 22,723 ( 80%)      3 
Split 5    24,148 (53.7%)     11.3%        0.53 25,607 ( 90%)      4 
Split 6    22,444 (49.9%)     11.1%        0.61 24,343 ( 86%)      3 
Split 7    43,865 (97.5%)      3.6%        0.16 22,199 ( 78%)      3 
Resample   43,865 (97.5%)       NA%          NA 42,197 (148%)     NA 

Sampling diagnostics for SMC run 7 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,868 (55.3%)     10.4%        0.85 28,242 ( 99%)      8 
Split 2    28,847 (64.1%)     12.5%        0.50 24,566 ( 86%)      5 
Split 3    17,950 (39.9%)     14.6%        0.69 25,834 ( 91%)      4 
Split 4    30,485 (67.7%)     12.7%        0.54 22,936 ( 81%)      4 
Split 5    23,884 (53.1%)     15.0%        0.52 25,517 ( 90%)      3 
Split 6    22,645 (50.3%)     15.8%        0.60 24,435 ( 86%)      2 
Split 7    43,935 (97.6%)      5.4%        0.16 21,948 ( 77%)      2 
Resample   43,935 (97.6%)       NA%          NA 42,158 (148%)     NA 

Sampling diagnostics for SMC run 8 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,878 (55.3%)     14.0%        0.85 28,341 (100%)      6 
Split 2    29,024 (64.5%)     12.6%        0.50 24,537 ( 86%)      5 
Split 3    17,830 (39.6%)     14.5%        0.69 25,771 ( 91%)      4 
Split 4    30,680 (68.2%)     16.7%        0.54 22,726 ( 80%)      3 
Split 5    23,128 (51.4%)     11.2%        0.53 25,635 ( 90%)      4 
Split 6    22,602 (50.2%)     11.1%        0.60 24,235 ( 85%)      3 
Split 7    43,898 (97.6%)      3.6%        0.16 22,062 ( 78%)      3 
Resample   43,898 (97.6%)       NA%          NA 42,154 (148%)     NA 

Sampling diagnostics for SMC run 9 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,888 (55.3%)     12.0%        0.85 28,194 ( 99%)      7 
Split 2    29,216 (64.9%)     12.5%        0.50 24,534 ( 86%)      5 
Split 3    17,954 (39.9%)     14.5%        0.68 25,930 ( 91%)      4 
Split 4    30,212 (67.1%)     16.8%        0.54 22,863 ( 80%)      3 
Split 5    24,084 (53.5%)     14.9%        0.52 25,508 ( 90%)      3 
Split 6    22,550 (50.1%)     15.9%        0.60 24,356 ( 86%)      2 
Split 7    43,908 (97.6%)      5.3%        0.16 21,934 ( 77%)      2 
Resample   43,908 (97.6%)       NA%          NA 42,210 (148%)     NA 

Sampling diagnostics for SMC run 10 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    24,880 (55.3%)     13.9%        0.84 28,311 (100%)      6 
Split 2    29,642 (65.9%)     12.6%        0.49 24,493 ( 86%)      5 
Split 3    17,844 (39.7%)     14.4%        0.69 25,903 ( 91%)      4 
Split 4    30,477 (67.7%)     16.7%        0.54 22,849 ( 80%)      3 
Split 5    23,225 (51.6%)     14.8%        0.53 25,558 ( 90%)      3 
Split 6    22,200 (49.3%)     16.0%        0.61 24,391 ( 86%)      2 
Split 7    43,945 (97.7%)      5.3%        0.15 22,057 ( 78%)      2 
Resample   43,945 (97.7%)       NA%          NA 42,243 (149%)     NA 

•  Watch out for low effective samples, very low acceptance rates (less than
1%), large std. devs. of the log weights (more than 3 or so), and low numbers
of unique plans. R-hat values for summary statistics should be between 1 and
1.05.
```
